File: di-string-match/di-string-match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        vector<int> ans;
        int i = 0;
        int d = s.length();
        for(auto it : s){
            if(it=='I'){ans.push_back(i); i++;}
            else{ans.push_back(d); d--;};
        }
        ans.push_back(i);
        return ans;
    }
};

File: di-string-match/di-string-match.java
class Solution {
    public int[] diStringMatch(String s) {
        List<Integer> ans = new ArrayList<>();
        int i = 0;
        int d = s.length();
        for (char it : s.toCharArray()) {
            if (it == 'I') {
                ans.add(i);
                i++;
            } else {
                ans.add(d);
                d--;
            }
        }
        ans.add(i);

        int[] result = new int[ans.size()];
        for (int j = 0; j < ans.size(); j++) {
            result[j] = ans.get(j);
        }

        return result;
    }
}
